### PR TITLE
create framework.api.common

### PR DESCRIPTION
**Pull request by techcobweb**
Signed-off-by: Mike Cobbett <77053+techcobweb@users.noreply.github.com>

- [x] Moved common classes into separate OSGi bundle dev.galasa.framework.api.common
- [x] BaseServlet is Ras-specific, so re-named it to RasServlet
- [x] ResponseBuilder introduced to replace the function called from routes and servlets...
- [x] added getResponseBuilder() calls to BaseRoute and to BaseServlet.
- [x] Updated versions of plugins
- [x] added dependencies between projects
- [x] OSGi bnd.bnd files updated with imports and exports
- [x] Log a warning in the error log if we can't write the payload contents of the HTTP response. Was being ignored silently.
- [x] Moved functions which were clumping together in RasRoute down to leaf-routes.
- [x] Stop every route setting the framework in the parent - dangerous.
- [x] Added getters for filesystem and framework.

### CODE REVIEW

**Comment by galasa-team** [comment]
Build successful

**Comment by techcobweb** [comment]
@KirbyKatcher FYI ^^^

**Comment by galasa-team** [comment]
Build successful

**Comment by galasa-team** [comment]
Build successful

**Comment by galasa-team** [comment]
Build successful

**Comment by galasa-team** [comment]
Build successful

**Comment by galasa-team** [comment]
Build successful

**Comment by galasa-team** [comment]
Build successful

**Comment by galasa-team** [comment]
Build successful

**Comment by galasa-team** [comment]
Build successful

File: galasa-parent/dev.galasa.framework.api.common/src/main/java/dev/galasa/framework/api/common/Something.java
Comment (KirbyKatcher) [inline galasa-parent/dev.galasa.framework.api.common/src/main/java/dev/galasa/framework/api/common/Something.java:1]: This class doesn't seem to be doing anything. Is it needed?

File: galasa-parent/dev.galasa.framework/src/main/java/dev/galasa/framework/internal/rps/tree/RpsResolver.java
Comment (KirbyKatcher) [inline galasa-parent/dev.galasa.framework/src/main/java/dev/galasa/framework/internal/rps/tree/RpsResolver.java:1]: This is showing as missing the license comment, do we need it or is this going to be over-written by another build?

**Review by KirbyKatcher — CHANGES_REQUESTED** [review]

**Comment by galasa-team** [comment]
Build successful

**Comment by galasa-team** [comment]
Build successful

**Comment by galasa-team** [comment]
Build successful

**Comment by galasa-team** [comment]
Build successful

**Comment by galasa-team** [comment]
Build successful

**Review by eamansour — APPROVED** [review]

**Comment by galasa-team** [comment]
Build successful

**Pull request merged**